### PR TITLE
ci-operator: Remove the OCP e2e-related jobs from the upstream OLM/registry prow configurations

### PR DESCRIPTION
**Pull request by timflannagan**
Update the upstream OLM prow configuration and remove any of the
OCP e2e-related jobs for the master/release-4.8+ branches.

Remove the prow configuration for the release-4.8+ branches in the
operator-framework/operator-registry component.

That unit test prow job is a duplicate as there's an existing GH action
in the upstream repository that covers that same check.

Remove that integration/e2e check with the latest OCP OLM images as we
can transition that work to live as a GH action at a later date.

### CODE REVIEW

File: ci-operator/config/operator-framework/operator-lifecycle-manager/operator-framework-operator-lifecycle-manager-master.yaml [hunk 1]
```diff
@@ -51,38 +51,6 @@ tests:
   commands: make unit
   container:
     from: src
```
Comment (timflannagan) [inline ci-operator/config/operator-framework/operator-lifecycle-manager/operator-framework-operator-lifecycle-manager-master.yaml:53]: I need to double-check whether this already lives as a GH action in the upstream OLM repository.

Comment (kevinrizza) [reply ci-operator/config/operator-framework/operator-lifecycle-manager/operator-framework-operator-lifecycle-manager-master.yaml:53]: It does

Comment (timflannagan) [reply ci-operator/config/operator-framework/operator-lifecycle-manager/operator-framework-operator-lifecycle-manager-master.yaml:53]: Including that unit test job, here are the only remaining jobs in the master/4.8+ branch prow configurations:

```
- as: verify
  commands: make verify
  container:
    from: src
- as: unit
  commands: make unit
  container:
    from: src
```

Let's get this in first, and I can look into ensuring those jobs live as GH actions before removing the master/release-4.8+ prow configuration entirely for upstream OLM.

**Pull request merged**